spec/facility_spec.rb

```diff
@@ -2,25 +2,171 @@
+    it 'returns plate_type category' do
+      # require 'pry'; binding.pry
+      expect(@cruz.plate_type).to eq(:regular)
+    end
+  end
+
+  describe '#collected_fees'do
+    it 'collects car registration fees' do
+      expect(@facility_1.register_vehicle(@cruz)).to eq([@cruz])
+      expect(@facility_1.registered_vehicles).to eq([@cruz])
+      expect(@facility_1.rcollected_fees(@cruz)).to eq(100)
+    
+      expect(@facility_1.register_vehicle(@camaro)).to eq([@cruz, @camaro])
+      expect(@camaro.new_registration_date).to eq(@camaro.new_registration_date)#check this out soon
+      expect(@facility_1.registered_vehicles).to eq([@cruz, @camaro])
+      expect(@facility_1.rcollected_fees(@camaro)).to eq(125) 
+      expect(@camaro.plate_type).to eq(:antique)
+
+      expect(@facility_1.register_vehicle(@bolt)).to eq([@cruz, @camaro, @bolt])
+      expect(@bolt.new_registration_date).to eq(@bolt.new_registration_date)#check this out soon
+      expect(@bolt.plate_type).to eq(:ev)
+      expect(@facility_1.registered_vehicles).to eq([@cruz, @camaro, @bolt])
+      expect(@facility_1.rcollected_fees(@bolt)).to eq(325) 
+    end
+
+    it ' registeres vehicles in the 2nd facility' do
+      expect(@facility_2.registered_vehicles).to eq([])
+      expect(@facility_2.services).to eq([])
+      expect(@facility_2.collected_fees).to eq(0) 
+      @facility_2.register_vehicle(@bolt)#interaction pattern says nil
+      expect(@facility_2.registered_vehicles).to eq([@bolt])
+      expect(@facility_2.rcollected_fees(@bolt)).to eq(200) 
+    end
+  end
+
+  describe "written test" do
+    it "gives exam to registrant" do
+      expect(@registrant_1.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+      expect(@registrant_1.permit?).to eq(true)
+      expect(@facility_1.administer_written_test(@registrant_1)).to eq(false)
+      expect(@registrant_1.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+      @facility_1.add_service('Written Test')
+      expect(@facility_1.administer_written_test(@registrant_1)).to eq(true)
+      expect(@registrant_1.license_data).to eq({:written=>true, :license=>false, :renewed=>false})
+
+      expect(@registrant_2.age).to eq(16)
+      expect(@registrant_2.permit?).to eq(false)
+      expect(@facility_1.administer_written_test(@registrant_2)).to eq(false)
+      @registrant_2.earn_permit
+      expect(@facility_1.administer_written_test(@registrant_2)).to eq(true)
+      expect(@registrant_2.license_data).to eq({:written=>true, :license=>false, :renewed=>false})
+      
+      expect(@registrant_3.age).to eq(15)
+      expect(@registrant_3.permit?).to eq(false)
+      expect(@facility_1.administer_written_test(@registrant_3)).to eq (false)
+      @registrant_3.earn_permit
+      expect(@facility_1.administer_written_test(@registrant_3)).to eq (false)
+      expect(@registrant_3.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+    end
```

Comment:
Nice job thoroughly testing the different scenarios in calling `administer_written_test`. You could keep all of these and separate them into individual `it` blocks for readability:
```
it 'does not administer written test when facility does not offer the service' do
  ...
end

it 'administers a written test to an eligible registrant when the facility offers the service' do
  ...
end

it 'does not administer written test to registrants without a permit' do
  ...
end

it 'does not administer written test to registrants under 16' do
  ...
end
```
Alternatively, you could organize these using additional `describe` blocks
```
describe '#administer_written_test' do
  describe 'to eligible registrants' do
    it 'administers a written test' do
      ...
    end
  end

  describe 'to ineligible registrants' do
    it 'does not administer written test if registrant has no permit' do
      ...
    end

    it 'does not administer written test if registrant is under 16' do
      ...
    end
  end
end
```